fstar-helpers/core-models/src/abstractions/bitvec.rs

```diff
@@ -300,43 +309,53 @@ pub mod int_vec_interp {
                 #[doc = concat!(stringify!($ty), " vectors of size ", stringify!($m))]
                 #[allow(non_camel_case_types)]
                 pub type $name = FunArray<$m, $ty>;
-                const _: ()  = {
-                    #[doc = concat!("Conversion from bit vectors of size ", stringify!($n), " to ", stringify!($ty), " vectors of size ", stringify!($m))]
-                    #[hax_lib::opaque]
-                    impl From<BitVec<$n>> for $name {
-                        fn from(bv: BitVec<$n>) -> Self {
-                            let vec: Vec<$ty> = bv.to_vec();
-                            Self::from_fn(|i| vec[i as usize])
+                pastey::paste! {
```

Comment:
I don't think this is good and readable code.

It seems the only reason you need `pastey` here is because you want to concatenate names. But why do you need those names? And how much additional code would you have to write without?